src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java

```diff
@@ -285,6 +325,80 @@ private static void verifyAstRaw(String expectedTextPrintFileName, String actual
                 "Generated AST from Java code should match pre-defined AST");
     }
 
+    private static final class AssertGeneratedTextBlockLexer
+            extends GeneratedTextBlockLexer {
+
+        private int laPosition;
+        private char[] laResults;
+
+        private AssertGeneratedTextBlockLexer() {
+            super((InputStream) null);
+        }
+
+        public static void verifyPass(String methodName, char... laResults) throws Exception {
+            verify(methodName, true, 1, laResults);
+        }
+
+        public static void verifyFail(String methodName, char... laResults) throws Exception {
+            verify(methodName, false, 1, laResults);
+        }
+
+        private static void verify(String methodName, boolean expectPass, int guessing,
+                                   char... laResults) throws Exception {
+            final AssertGeneratedTextBlockLexer instance =
+                    new AssertGeneratedTextBlockLexer();
+            instance.laPosition = 0;
+            instance.laResults = laResults.clone();
+            instance.inputState.guessing = guessing;
+
+            final Method method = GeneratedTextBlockLexer.class.getDeclaredMethod(methodName,
```

Comment:
I would rather do no coverage for such hard to catch code and and bump percentage in config.
we keep percentage in coverage for generated code only to see diff of coverage on new grammar update and probably discuss what could be covered by real inputs.
We do not need to have test on generated code of ANTLR, it is by definition the most trusty code in our code base.

Reply:
ok, looks like @rnveach  was a person who introduced such asserts - https://github.com/checkstyle/checkstyle/commit/5123ab33a6cbb67c5e5c2d1dd99f1bb39180e100 .
ok, lets keep it.